### PR TITLE
update package to use pyproject.toml 

### PR DESCRIPTION
**Pull request by bandophahita**
and be more in sync with screenpy repo

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,31 @@
+[flake8]
+exclude =
```
Comment (perrygoy) [inline .flake8:2]: After seeing this list for the third time, does `flake8` actually try to parse these non-.py files?

File: docs/conf.py [hunk 2]
```diff
@@ -78,5 +78,4 @@
 html_static_path = ['_static']
 
 # Other HTML settings
-autodoc_member_order = "bysource"
```
Comment (perrygoy) [inline docs/conf.py:81]: Why was this one removed? This defaults to alphabetical order, which i think makes it more confusing in the documentation.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+# But then you need to remember to manually edit the [tool.poetry.extras] dev section
+# and declare the package. This allows `pip install .[dev]` to work as expected
+# Poetry users will need to use the `--extras dev` option rather than the `--with dev`
+# so we dont have two different sets of package versions to update.
+
+[tool.poetry.dependencies]
+python = "^3.8"
+screenpy = ">=4.0.2"
+requests = ">=2.26.0"
+typing-extensions = ">=4.1.1"
```
Comment (perrygoy) [inline pyproject.toml]: I don't think we use `typing-extensions` in this one, either. I think it's just ScreenPy itself that uses it.

**Review by perrygoy — COMMENTED** [review]
A couple small questions that i'd like to discuss before approving, but again—thank you for all this standardization!

File: .flake8 (hunk 1 above)
Comment (bandophahita) [reply .flake8:2]: It can, depending on how you run it.  This avoids the possibility of it trying.

File: docs/conf.py (hunk 2 above)
Comment (bandophahita) [reply docs/conf.py:81]: There were two of them in the file.

File: pyproject.toml (hunk 3 above)
Comment (bandophahita) [reply pyproject.toml]: I just moved everything from the setup.py into this.  I didn't bother checking if it was actually needed.

File: docs/conf.py (hunk 2 above)
Comment (perrygoy) [reply docs/conf.py:81]: Haha, oh. That'll do it.

File: pyproject.toml (hunk 3 above)
Comment (perrygoy) [reply pyproject.toml]: This would be a good opportunity to fix that in both places!

**Review by perrygoy — APPROVED** [review]
Woo! Thanks, @bandophahita!

**Pull request merged**